Support Testing Integration with secretgen-controller

danielhelfand opened the pull request:
Closes #282 

This pull request adds the ability to deploy secretgen-controller and run associated tests for kapp-controller's integration with secretgen-controller based on an environment variable being set named `KAPPCTRL_E2E_SECRETGEN-CONTROLLER`. 

If `KAPPCTRL_E2E_SECRETGEN-CONTROLLER=true`, both deploy.sh and deploy-test.sh will deploy a specific version of secretgen-controller. Additionally, if `KAPPCTRL_E2E_SECRETGEN-CONTROLLER=true`, test-all.sh will run tests under the secretgencontroller folder.

danielhelfand commented:
Just a note to anyone reviewing this that GitHub actions are experiencing issues at the moment: https://www.githubstatus.com/. We should let CI run before introducing this if reviews look good.

cppforlife commented on .github/workflows/test-gh.yml:
```diff
@@ -51,6 +51,7 @@ jobs:
         minikube start --driver=docker
         eval $(minikube docker-env --shell=bash)
 
+        export SECRETGEN-CONTROLLER=true
```
i think we should prefix this env var with KAPPCTRL_E2E_. lets also use _ instead of -

danielhelfand replied:
Thanks, added

danielhelfand committed:
move tests to folders based on whether secretgen-controller must be deployed for tests

All tests not requiring secretgen-controller will be under a folder called kappcontroller. All those requiring secretgen-controller will exist under a folder called secretgencontroller.

danielhelfand committed:
add KAPPCTRL_E2E_SECRETGEN_CONTROLLER env var to determine whether to deploy/run tests

If KAPPCTRL_E2E_SECRETGEN_CONTROLLER=true, secretgen-controller will be deployed to cluster with kapp-controller and associated tests will be run

danielhelfand committed:
set KAPPCTRL_E2E_SECRETGEN_CONTROLLER env var in CI

danielhelfand committed:
update documentation to mention KAPPCTRL_E2E_SECRETGEN_CONTROLLER
